lib/OpenQA/Task/Job/Limit.pm

```diff
@@ -65,6 +67,8 @@ sub _limit {
         }
     }
 
+    conclude_on_signals;
```

Comment:
It would probably be safer if this also used `$job` directly and did a `$job->finish`. Otherwise you'd leave that call to the parent process, and you can't be sure in what state that process is. Less race conditions if the job process updates the status itself.